File: src/com/hkb48/keepdo/CheckSoundPlayer.java
package com.hkb48.keepdo;

import android.content.Context;
import android.media.AudioManager;
import android.media.SoundPool;

class CheckSoundPlayer {
    private final Context mContext;
    private SoundPool mSoundPool;
    private int mSoundId;

    CheckSoundPlayer(Context context) {
        mContext = context;
    }

    void load() {
        mSoundPool = new SoundPool(1, AudioManager.STREAM_SYSTEM, 0);
        mSoundId = mSoundPool.load(mContext, R.raw.done_pressed, 1);
    }

    void unload() {
        mSoundPool.release();
    }

    void play() {
        mSoundPool.play(mSoundId, 1.0f, 1.0f, 0, 0, 1.0f);
    }
}

File: app/src/main/java/com/hkb48/keepdo/CheckSoundPlayer.kt
package com.hkb48.keepdo

import android.content.Context
import android.media.SoundPool
import com.hkb48.keepdo.util.CompatUtil

class CheckSoundPlayer(private val context: Context) {
    private var soundPool: SoundPool? = null
    private var soundId = 0
    fun load() {
        soundPool = CompatUtil.soundPool
        soundId = soundPool!!.load(context, R.raw.done_pressed, 1)
    }

    fun unload() {
        soundPool!!.release()
    }

    fun play() {
        soundPool!!.play(soundId, 1.0f, 1.0f, 0, 0, 1.0f)
    }
}